### ISSUE TITLE
1.9.1 undefined symbols

### ISSUE THREAD

**Issue by StevoSM** [DevMate/DevMateKit#47]
We just updated to DevMateKit 1.9.1 and when building in Xcode, we are seeing the following error:

Undefined symbols for architecture x86_64:
  "DMTrialHasTrialHistory(__DMTrial*)", referenced from:
      ____ZL22DMKitDebugAddTrialMenuv_block_invoke_3 in MTAppDelegate.o
  "DMTrialInvalidateTrial(__DMTrial*)", referenced from:
      ____ZL22DMKitDebugAddTrialMenuv_block_invoke in MTAppDelegate.o

**Comment by dimaty** [comment]
It's very strange. Is it possible that you're using standard and MAS version of DevMateKit in your project and now for some reason your target started linking with MAS version that doesn't contain activation (and trial) functionality?

**Comment by StevoSM** [comment]
That doesn't sound very likely.

Everything was linking fine previously.  Then I did a pod update which updated from v1.8.2 to 1.9.1, now I'm getting the linking errors.

The linking errors come when I add the DMKitDebugAddDevMateMenu() call to the application.  Alternatively, if instead I add DMKitDebugAddActivationMenu(), everything links fine, but if I add DMKitDebugAddTrialMenu() I again get the linking errors.

**Comment by dimaty** [comment]
Thanks for more information. I’ll recheck it with pod version.

**Comment by djogon** [comment]
Just to chime in and confirm that this is happening here as well. Exact same issue.

```
Undefined symbols for architecture x86_64:
  "DMTrialHasTrialHistory(__DMTrial*)", referenced from:
      ____ZL22DMKitDebugAddTrialMenuv_block_invoke_3 in DevMateKitWrapper.o
  "DMTrialInvalidateTrial(__DMTrial*)", referenced from:
      ____ZL22DMKitDebugAddTrialMenuv_block_invoke in DevMateKitWrapper.o
ld: symbol(s) not found for architecture x86_64
```

Basically – inside the DevMateInlines.h file there is a function called 
`DM_INLINE void DMKitDebugAddTrialMenu(void)`

    it wants to call DMTrialHasTrialHistory(NULL); which the linker cannot find. Same goes for the “DMTrialInvalidateTrial” function.

**Comment by dimaty** [comment]
Ok, I've just created new test project, added podfile with DevMateKit dependency, in `applicationDidFinishLaunching` added call of funcs `DMKitDebugAddDevMateMenu` and `DMTrialHasTrialHistory` and everything was built without any errors (checked it with ObjC codebased target and Swift target). Could you do the same: create test project and try to reproduce?

**Comment by StevoSM** [comment]
When I create a sample project, if the file is "AppDelegate.m", I get no errors, but if I change the filename to "AppDelegate.mm", I get the errors.  We are using ObjC++ in the files that are accessing DevMateKit.

**Comment by dimaty** [comment]
Now I confirm that linker error exist with ObjC++ codebase

**Comment by dimaty** [comment]
Please, check it with released version 1.9.2.

**Comment by StevoSM** [comment]
Looks like it's all good now.  Thanks!
